src/OpenTelemetry.AutoInstrumentation.Native/logger_impl.h

```diff
@@ -131,28 +138,41 @@ LoggerImpl<TLoggerPolicy>::LoggerImpl()
 
     spdlog::flush_every(std::chrono::seconds(3));
 
-    static auto current_process_name = ToString(GetCurrentProcessName());
-    static auto current_process_id   = GetPID();
-    static auto current_process_without_extension =
-        current_process_name.substr(0, current_process_name.find_last_of("."));
+    static auto configured_log_sink = GetLogSinkConfiguration();
+
+    if (configured_log_sink == log_sink_none)
+    {
+        m_fileout = spdlog::null_logger_mt(logger_name);
+    }
+    else if (configured_log_sink == log_sink_file)
+    {
+        static auto current_process_name = ToString(GetCurrentProcessName());
+        static auto current_process_id   = GetPID();
+        static auto current_process_without_extension =
+            current_process_name.substr(0, current_process_name.find_last_of("."));
 
-    static auto file_name_suffix =
-        std::to_string(current_process_id) + "-" + current_process_without_extension + "-Native";
+        static auto file_name_suffix =
+            std::to_string(current_process_id) + "-" + current_process_without_extension + "-Native";
 
-    // by default, use the same size as on managed side: 10MiB
-    static auto file_size        = GetConfiguredSize(environment::max_log_file_size, 10485760);
+        // by default, use the same size as on managed side: 10MiB
+        static auto file_size = GetConfiguredSize(environment::max_log_file_size, 10485760);
 
-    try
-    {
-        m_fileout = spdlog::rotating_logger_mt(logger_name, GetLogPath(file_name_suffix), file_size, 10);
+        try
+        {
+            m_fileout = spdlog::rotating_logger_mt(logger_name, GetLogPath(file_name_suffix), file_size, 10);
+        }
+        catch (...)
+        {
+            // By writing into the stderr was changing the behavior in a CI scenario.
+            // There's not a good way to report errors when trying to create the log file.
+            // But we never should be changing the normal behavior of an app.
+            // std::cerr << "LoggerImpl Handler: Error creating native log file." << std::endl;
+            m_fileout = spdlog::null_logger_mt(logger_name);
+        }
     }
-    catch (...)
+    else if (configured_log_sink == log_sink_console)
     {
-        // By writing into the stderr was changing the behavior in a CI scenario.
-        // There's not a good way to report errors when trying to create the log file.
-        // But we never should be changing the normal behavior of an app.
-        // std::cerr << "LoggerImpl Handler: Error creating native log file." << std::endl;
-        m_fileout = spdlog::null_logger_mt(logger_name);
+        m_fileout = spdlog::stdout_logger_mt(logger_name);
     }
```

Comment:
it should log a warning if didn't matching any and fallback to the default (file).

Reply:
Something seems off regarding to this comment and diff 🤔
Could you check it again after some refactoring was done?

Reply:
`configured_log_sink` is passed to this function with whatever value was on the environment variable, defaulting to `log_sink_file` if the environment variable was not specified, it seems that `m_fileout` is not set to an explicit fallback, and just reading the code around here I can't tell what happens in this case. Typically for log mechanisms we fallback to the "default" one if the configuration is incorrect. I'm opening to considering changing the fallback to console instead of file. That said reading just the code around here I can't tell what will happen if there was a typo on the value of the environment variable.

Reply:
> That said reading just the code around here I can't tell what will happen if there was a typo on the value of the environment variable.

Ah that's a good point, entirely missing. I need to refactor that.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3558/commits/9b042ce36291835079025a8d1c613ec594e958ea